spec/uber_task/logger_spec.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+describe UberTask do
+  before do
+    described_class.instance_variable_set(:@logger, nil)
+  end
+
+  describe '.logger' do
+    it 'memoizes variable' do
+      initialized_logger = described_class.logger
+
+      expect(described_class.logger.object_id).to eq(
+        initialized_logger.object_id,
+      )
+    end
+
+    context 'when Rails logger is not defined' do
```

Comment:
Next PR I'm working on will introduce tests in Rails application environment and `context 'when Rails logger is defined'` will be added